python_scripts/libaerie/products/product_parser.py

```diff
@@ -233,10 +380,18 @@ def parse(self):
             num_r += 1
             yield r
 
-        logger.info("Got %s activites from %s", num_r, self._fh.name)
+        logger.info("Got %s activites from %s", num_r, self.filename)
 
     @classmethod
     def rtlt_to_timedelta(cls, rtlt_dur_str: str) -> datetime.timedelta:
+        """
+        Converts the formatted duration from the Viewperiod RTLT field into a python timedelta
```

Comment:
I'm getting a little deja vu reading this function. Have you considered merging the two similar function definitions? Or is it best to keep them separate?

Reply:
Haha I think your Deja Vu is Deja True, I'll move this definition to the Decoder parent class.